src/plugins/kibana/public/settings/sections/advanced/lib/is_immutable_config.js

```diff
@@ -0,0 +1,15 @@
+define(function (require) {
+  var _ = require('lodash');
+
+  var IMMUTABLE_CONFIG_VALS = ['buildNum'];
```

Comment:
I'm not sure if there are any other fields in .kibana/config that we should whitelist.  `buildNum` was the only one in my local setup.